mod/search/start.php

```diff
@@ -524,3 +512,41 @@ function search_exclude_robots($hook, $type, $text) {
 
 	return $text;
 }
+
+/**
+ * Returns minimum and maximum lengths of words for MySQL search
+ * This function looks for stored config values, and, if none set,
+ * queries the DB and saves them
+ * @return stdClass An object with min and max properties
+ */
+function search_get_ft_min_max() {
+
+	$min = (int) elgg_get_config('search_ft_min_word_len');
+	$max = (int) elgg_get_config('search_ft_max_word_len');
+
+	if (!$min || !$max) {
+		try {
+			$result = get_data_row('SELECT @@ft_min_word_len as min, @@ft_max_word_len as max');
+		} catch (DatabaseException $e) {
+			// some servers don't have these values set which leads to exception
+			// we ignore the exception
+		}
+		if ($result) {
```

Comment:
If the exception was thrown, $result won't be defined so this will cause a notice. Maybe move the assignments to the try and catch blocks?